xen/include/public/io/sndif.h

```diff
@@ -277,143 +398,157 @@
+ * Request set/get volume - set/get channels' volume of the stream given:
+ *          0                 1                  2                3        octet
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                 id                |    operation    |     stream_idx  |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                              reserved                                 |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/|
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                              reserved                                 |
+ * +-----------------+-----------------+-----------------+-----------------+
  *
- * Shared page for set/get volume:
- *
- *     0    1     2     3     4     5     6     7  octet
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |        vol_ch0        |        vol_ch1        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |        vol_ch2        |        vol_ch3        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/+
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |       vol_ch126       |       vol_ch127       |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * id - private guest value, echoed in resp
+ * operation - XENSND_OP_SET_VOLUME or XENSND_OP_GET_VOLUME
+ * stream_idx - index of the stream ("streams_idx" XenStore entry
+ *   of the stream)
+ * gref[0] is used to exchange volume values from the open request.
+ *
+ * Shared page for set/get volume data:
+ *          0                 1                  2                3        octet
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                                vol_ch0                                |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                                vol_ch1                                |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/|
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                                vol_ch127                              |
+ * +-----------------+-----------------+-----------------+-----------------+
  *
  * vol_ch0 - vol_ch127 - volume for the channel from 0 to
  *   XENSND_MAX_CHANNELS_PER_STREAM
- * Please, note that only first 'stream%u_channels' are used in this command,
- *   where 'stream%u_channels' is read from the XenStore (channels count for
- *   stream with index '%u' which equals to 'stream_idx')
+ * Volume is expressed as a signed value in steps of 0.001 dBm,
+ * while 0 is being 0dBm.
+ *
+ *
+ * Request mute/unmute - mute/unmute stream:
```

Comment:
We could have sync issue between FE and BE: in some reason on BE stream could be unmuted but FE considering them as muted and intends to unmute but BE will mute instead. I think better to have separate commands (or special field to indicate mute or unmute) and track the mute state inside FE.